extensions-core/caffeine-cache/src/main/java/io/druid/client/cache/CaffeineCache.java

```diff
@@ -0,0 +1,198 @@
+
+
+  public static CaffeineCache create(final CaffeineCacheConfig config)
+  {
+    return create(config, config.createExecutor());
+  }
+
+  // Used in testing
+  public static CaffeineCache create(final CaffeineCacheConfig config, @Nullable final Executor executor)
+  {
+    Caffeine<Object, Object> builder = Caffeine.newBuilder().recordStats();
+    if (config.getExpireAfter() >= 0) {
+      builder
+          .expireAfterAccess(config.getExpireAfter(), TimeUnit.MILLISECONDS);
+    }
+    if (config.getSizeInBytes() >= 0) {
+      builder
+          .maximumWeight(config.getSizeInBytes())
+          .weigher((NamedKey key, byte[] value) -> value.length
+                                                   + key.key.length
+                                                   + key.namespace.length() * Chars.BYTES
+                                                   + FIXED_COST);
+    }
+    if (executor != null) {
+      builder.executor(executor);
+    }
+    return new CaffeineCache(builder.build(), config);
+  }
+
+  private CaffeineCache(final Cache<NamedKey, byte[]> cache, CaffeineCacheConfig config)
+  {
+    this.cache = cache;
+    this.config = config;
+  }
+
+  @Override
+  public byte[] get(NamedKey key)
+  {
+    return deserialize(cache.getIfPresent(key));
+  }
+
+  @Override
+  public void put(NamedKey key, byte[] value)
+  {
+    cache.put(key, serialize(value));
+  }
+
+  @Override
+  public Map<NamedKey, byte[]> getBulk(Iterable<NamedKey> keys)
+  {
+    // The assumption here is that every value is accessed at least once. Materializing here ensures deserialize is only
+    // called *once* per value.
+    return ImmutableMap.copyOf(Maps.transformValues(cache.getAllPresent(keys), this::deserialize));
+  }
+
+  // This is completely racy with put. Any values missed should be evicted later anyways. So no worries.
+  @Override
+  public void close(String namespace)
+  {
+    if (config.isEvictOnClose()) {
+      cache.asMap().keySet().removeIf(key -> key.namespace.equals(namespace));
+    }
+  }
+
+  @Override
+  public io.druid.client.cache.CacheStats getStats()
+  {
+    final com.github.benmanes.caffeine.cache.stats.CacheStats stats = cache.stats();
+    final long size = cache
+        .policy().eviction()
+        .map(eviction -> eviction.isWeighted() ? eviction.weightedSize() : OptionalLong.empty())
+        .orElse(OptionalLong.empty()).orElse(-1);
+    return new io.druid.client.cache.CacheStats(
+        stats.hitCount(),
+        stats.missCount(),
+        cache.estimatedSize(),
+        size,
+        stats.evictionCount(),
+        0,
+        stats.loadFailureCount()
+    );
+  }
+
+  @Override
+  public boolean isLocal()
+  {
+    return true;
+  }
+
+  @Override
+  public void doMonitor(ServiceEmitter emitter)
+  {
+    final CacheStats oldStats = priorStats.get();
+    final CacheStats newStats = cache.stats();
+    final CacheStats deltaStats;
+    if (oldStats == null) {
```

Comment:
You could initialize `oldStats` to `CacheStats.empty()`. Then if/else could be removed `newStats.minus(oldStats)`.